### DISCUSSION TITLE
Caching for SSR (With Cache API)

### DISCUSSION THREAD

**Discussion by Mustack** [vercel/vercel#4880] [Ideas] [answered]
I'm looking for a caching solution for my SSR'd pages. I think this could be useful in general but I'll describe my use-case. I'm working on [an event planning app similar to Facebook Events, called DirectEvents](https://directevents.io/) where users can create pages for their events. Because the event pages are user-generated, I can't build them statically. Some data like comments change often but other data like the event's title and description do not change often. I'd like to render a page with the details of the event, serve that quickly (ideally from a cache), and load comments on the client after the page is hydrated. It would be great if Vercel could cache that server-rendered page. Furthermore, if there was an API for the cache, it would also be very easy for me to tell it when to invalidate a page on the rare occasion that a user changes event details.

I know this is probably a lot to ask, especially from someone on a free tier, but this is a problem I'm exploring and thinking about so I wanted to share it with you in case you find it of value.

Thanks for building NextJS and Vercel! They're a real pleasure to work with.

**Comment by olpeh** [comment, marked as answer]
Quite late, but maybe someone else is looking for the answer. Set the cache control headers, so Vercel CDN will handle the caching: https://nextjs.org/docs/going-to-production#caching
